cunumeric/coverage.py

```diff
@@ -12,91 +12,184 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+from __future__ import annotations
 
-import inspect
 import warnings
+from functools import wraps
+from types import FunctionType, MethodDescriptorType, MethodType, ModuleType
+from typing import Any, Callable, Container, Optional
 
 from .runtime import runtime
 from .utils import find_last_user_frames, find_last_user_stacklevel
 
+__all__ = ("clone_module",)
+
+FALLBACK_WARNING = (
+    "cuNumeric has not implemented {name} "
+    + "and is falling back to canonical numpy. "
+    + "You may notice significantly decreased performance "
+    + "for this function call."
+)
+
+MOD_INTERNAL = {"__dir__", "__getattr__"}
+
+NDARRAY_INTERNAL = {
+    "__array_finalize__",
+    "__array_function__",
+    "__array_interface__",
+    "__array_prepare__",
+    "__array_priority__",
+    "__array_struct__",
+    "__array_ufunc__",
+    "__array_wrap__",
+}
```

Comment:
Without excluding these, there are errors doing conversions. I noticed they are not present on `cunumeric.ndarray` before this PR, so I omitted them. But not sure if this lists is comprehensive.

Reply:
I saw a similar behavior noted in the comment of the initial attempt to add `__array_interface__` to `cunumeric.ndarray`. Unless there's an evidence that this lis is incomplete, I'm fine with keeping this edit.